File: backend/src/main/java/com/smartlife/model/Note.java
package com.smartlife.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "notes")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Note {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User user;

    private String title;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @Column(columnDefinition = "TEXT[]")
    private String[] tags;

    private boolean isPinned = false;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() { updatedAt = LocalDateTime.now(); }
}

File: mobile-android/app/src/main/java/com/smartlife/mobile/feature/notes/data/model/Note.kt
package com.smartlife.mobile.feature.notes.data.model

data class Note(
    val id: Long = 0,
    val title: String = "",
    val content: String = "",
    val color: String? = null,
    val tags: List<String> = emptyList(),
    val pinned: Boolean = false,
    val createdAt: String = "",
)

data class CreateNoteRequest(
    val title: String,
    val content: String,
    val color: String? = null,
    val tags: List<String> = emptyList(),
)
